src/main/drivers/timer_hal.c

```diff
@@ -742,75 +753,157 @@ _TIM_IRQ_HANDLER(TIM1_UP_TIM16_IRQHandler, 16);    // only timer16 is used, not
 _TIM_IRQ_HANDLER(TIM1_TRG_COM_TIM17_IRQHandler, 17);
 #endif
 
-void timerInit(void)
+#define HAL_TIMER_CLOCK_ENABLE(t, n) if (t == TIM ## n) { __HAL_RCC_TIM ## n ## _CLK_ENABLE(); return; } 
+void timerRccEnable(TIM_TypeDef *timer)
 {
```

Comment:
It can I totally agree. This would be an area for cleanup once approach is agreed.